Index: lede/package/base-files/files/etc/config/system
===================================================================
--- lede.orig/package/base-files/files/etc/config/system	2016-05-30 00:49:04.378541545 +0200
+++ lede/package/base-files/files/etc/config/system	2016-05-30 00:49:40.867274745 +0200
@@ -1,5 +1,5 @@
 config system
-	option hostname	lede
+	option hostname	screwdriver
 	option timezone	UTC
 	option ttylogin	0
 
