Index: lede/target/linux/omap/modules.mk
===================================================================
--- /dev/null
+++ lede/target/linux/omap/modules.mk
@@ -0,0 +1,24 @@
+#
+# Copyright (C) 2010 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define KernelPackage/cape-manager
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Cape Manager
+  DEPENDS:=@TARGET_omap
+  KCONFIG:= \
+  	CONFIG_BONE_CAPEMGR
+  FILES:=\
+  	$(LINUX_DIR)/drivers/misc/bone_capemgr.ko
+  AUTOLOAD:=$(call AutoProbe,bone_capemgr)
+endef
+
+define KernelPackage/cape-manager/description
+  Beagle bone cape manager
+endef
+
+$(eval $(call KernelPackage,cape-manager))
+
Index: lede/target/linux/omap/config-4.4
===================================================================
--- lede.orig/target/linux/omap/config-4.4
+++ lede/target/linux/omap/config-4.4
@@ -69,7 +69,6 @@ CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=16
 CONFIG_BLK_DEV_RAM_SIZE=16384
 CONFIG_BLK_DEV_SD=y
-CONFIG_BONE_CAPEMGR=y
 CONFIG_BOUNCE=y
 CONFIG_BSD_PROCESS_ACCT=y
 CONFIG_BUILD_BIN2C=y
