Index: openwrt/package/boot/uboot-omap/Makefile
===================================================================
--- openwrt.orig/package/boot/uboot-omap/Makefile	2015-10-19 18:56:16.411090659 +0200
+++ openwrt/package/boot/uboot-omap/Makefile	2015-10-19 18:57:16.640300048 +0200
@@ -22,6 +22,7 @@
 PKG_LICENSE_FILES:=Licenses/README
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/image.mk
 
 define uboot/Default
   TITLE:=
@@ -85,6 +86,9 @@
 	$(INSTALL_DIR) $(BIN_DIR)/uboot-$(BOARD)-$(1)
 	$(CP) $(PKG_BUILD_DIR)/u-boot.img $(BIN_DIR)/uboot-$(BOARD)-$(1)/u-boot.img
 	$(CP) $(PKG_BUILD_DIR)/MLO $(BIN_DIR)/uboot-$(BOARD)-$(1)/MLO
+
+	$(CP) $(PKG_BUILD_DIR)/u-boot.img $(KDIR)/uboot-$(BOARD)-$(1)-u-boot.img
+	$(CP) $(PKG_BUILD_DIR)/MLO $(KDIR)/uboot-$(BOARD)-$(1)-MLO
 endef
 
 define Package/uboot/install/template
