Index: openwrt/package/base-files/files/etc/banner
===================================================================
--- openwrt.orig/package/base-files/files/etc/banner	2015-07-01 18:21:53.132832120 +0200
+++ openwrt/package/base-files/files/etc/banner	2015-12-09 20:10:26.583083387 +0100
@@ -1,14 +1,19 @@
-  _______                     ________        __
- |       |.-----.-----.-----.|  |  |  |.----.|  |_
- |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
- |_______||   __|_____|__|__||________||__|  |____|
-          |__| W I R E L E S S   F R E E D O M
+                               _      _
+                              | |    (_)
+ ___  ___ _ __ _____      ____| |_ __ ___   _____ _ __
+/ __|/ __| '__/ _ \ \ /\ / / _` | '__| \ \ / / _ \ '__|
+\__ \ (__| | |  __/\ V  V / (_| | |  | |\ V /  __/ |
+|___/\___|_|  \___| \_/\_/ \__,_|_|  |_| \_/ \___|_|
+
  -----------------------------------------------------
- CHAOS CALMER (%C, %R)
- -----------------------------------------------------
-  * 1 1/2 oz Gin            Shake with a glassful
-  * 1/4 oz Triple Sec       of broken ice and pour
-  * 3/4 oz Lime Juice       unstrained into a goblet.
-  * 1 1/2 oz Orange Juice
-  * 1 tsp. Grenadine Syrup
+ based on OpenWrt CHAOS CALMER (%C, %R)
  -----------------------------------------------------
+ To read the contents of a spi flash:
+ #> flashrom -p linux_spi:dev=/dev/spidev1.0 -r /tmp/content.rom
+
+ To write an image to the spi flash:
+ #> flashrom -p linux_spi:dev=/dev/spidev1.0 -w /tmp/newcontent.rom
+
+ To receive USB debug output:
+ #> cat /dev/ttyGS0 | tee -a /tmp/usbdebug
+
