Index: openwrt/package/base-files/files/etc/shadow
===================================================================
--- openwrt.orig/package/base-files/files/etc/shadow	2015-07-01 18:21:53.132832120 +0200
+++ openwrt/package/base-files/files/etc/shadow	2015-10-11 11:13:32.662530513 +0200
@@ -1,4 +1,4 @@
-root::0:0:99999:7:::
+root:$1$bAEkABgV$TR/ebAg3xU6UDzQs/o/uH0:0:0:99999:7:::
 daemon:*:0:0:99999:7:::
 ftp:*:0:0:99999:7:::
 network:*:0:0:99999:7:::
